### ISSUE TITLE
Load pretrained model

### ISSUE THREAD

**Issue by anzhi-L** [ultralytics/ultralytics#17135]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I tried to replace the backbone with PKINet and load the pre trained model of PKINet for training, but I found that the training results for each epoch were the same with and without the pre trained model, so I suspect that the pre trained model did not load successfully. We haven't found the problem yet, and we hope you can provide some help. thank you！

### Additional

```
from ultralytics import YOLO
import torch
model = YOLO("yolo11n-pkinet2.yaml")
ckpt1 = torch.load("pkinet_t_pretrain.pt")
ckpt = ckpt1['state_dict']
model_state_dict = model.state_dict()
m = []
for key in model_state_dict.keys():
    if int(key[12]) == 0:
        m.append(key)
    elif int(key[12]) == 1 and key[13] == ".":
        m.append(key)
        # print(key)
    elif int(key[12]) == 2 and key[13] == ".":
        m.append(key)
        # print(key)
    elif int(key[12]) == 3:
        m.append(key)
        # print(key)
    elif int(key[12]) == 4:
        m.append(key)
for i, (k, v) in enumerate(ckpt.items()):
    if i>=2610:
        break
    model_state_dict[m[i]] = v
model.load_state_dict(model_state_dict)
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @anzhi-L, thank you for reaching out and for your interest in Ultralytics 🚀! We understand that you are having questions about loading a pre-trained model with a custom backbone such as PKINet.

If this is a 🐛 bug report, please help us by providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) which allows us to investigate further. Your current snippet is a great start! Ensure you've included any necessary configuration files or additional steps to replicate your environment setup.

In the meantime, you might find some immediate insights by checking our comprehensive [Docs](https://docs.ultralytics.com). Look particularly at the [Python usage examples](https://docs.ultralytics.com/usage/python/) which might have similarities to your application. 

If you are new to custom backbone integration, reviewing the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) can be beneficial.

Our community platforms are also a great place for support and sharing insights:
- Join [Discord](https://ultralytics.com/discord) 🎧 for real-time conversations.
- Engage in extended threads on our [Discourse](https://community.ultralytics.com).
- Visit our [Subreddit](https://reddit.com/r/ultralytics) to exchange knowledge with others.

To ensure you have the latest updates, please upgrade your `ultralytics` package in your environment with:

```bash
pip install -U ultralytics
```

For your information, this is an automated response, but rest assured an Ultralytics engineer will review your issue shortly and provide additional assistance. Thanks for your patience 🙏!

Verify that you are using a compatible setup as outlined in the verified environments section below to rule out any setup-related causes:

YOLO may be run in these environments with pre-installed dependencies:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

For verified package integrity, check our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule), and ensure the green badge on the CI tests is active, verifying correct operations on multiple systems.

**Comment by Y-T-G** [comment]
It recreates the model from yaml and loads the weights that you loaded using `.load()` method. `state_dict()` updates would be ignored during training.

You can change this to `self.trainer.model = self.model` to use the `state_dict()` changes:

https://github.com/ultralytics/ultralytics/blob/be40a45ec32d9173ff42b4bdb00dbfe5a8c7a838/ultralytics/engine/model.py#L798

**Comment by anzhi-L** [comment]
Thank you, this issue has been resolved. Now facing a new problem, I have used pre-trained model parameters for the backbone section. Will setting a small learning rate affect the training of subsequent network layers? Can the lr for the backbone and neck sections be set separately?

**Comment by Y-T-G** [comment]
> Can the lr for the backbone and neck sections be set separately?

No. You can only freeze it.

**Comment by anzhi-L** [comment]
okay, thank you

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.
